File: 2021/May/Day 4/replace-digits.py
# Source : https://leetcode.com/problems/replace-all-digits-with-characters/submissions/
class Solution:
    def replaceDigits(self, s: str) -> str:
        prev =""

        new = ""
        for c in s:
            if(c.isalpha()):
                new = new +c;
            elif c.isdigit():
                new = new + chr(ord(prev)+int(c))

            prev = c;
        return new

File: 2021/August/Day 12/replaceDigits.cpp
class Solution
{
public:
    string replaceDigits(string s)
    {

        string n = "";
        char prev='a';
        for(int i =0;i<size(s);i++)
        {
            if(isalpha(s[i]))
            {n+=s[i];}
            if(isdigit(s[i]))
            {
                //cout<<char(int(s[i-1])+int(s[i]-48));
                n+=char(int(s[i-1])+int(s[i]-48));
            }
        }
        return n;
    }
};
